### DISCUSSION TITLE
restrict-asrockrack-bmc-to-ipmi-lan-port/

### DISCUSSION THREAD

**Discussion by giscus** [ironicbadger/blog.ktz.me#218] [Announcements]
# restrict-asrockrack-bmc-to-ipmi-lan-port/

During this process I came across a frustrating "bug" in the Asrock Rack BMC implementation. No matter the settings I gave the BMC it was getting two IP addresses. One on the IPMI_LAN port as expected in my management VLAN, and another on eth0 which is undesirable.

https://blog.ktz.me/restrict-asrockrack-bmc-to-ipmi-lan-port/

**Comment by BadCo-NZ** [comment]
I had this bug on a E3C224D2I being used as a gateway with pfsense. Every time the gateway was rebooted it would expose the IPMI webGUI to the internet on the WAN port.

I couldn't figure out how to fix it so moved the WAN port to the other ethernet port. Just applied your fix so hopefully that fixes it for the future, thanks @ironicbadger!

**Reply by ironicbadger** [reply]
Holy cow that's really really bad. Like never want to buy an ASRock board again bad. What were they thinking?!

**Reply by BadCo-NZ** [reply]
Hopefully it was a bug and not deliberate 😅

The IPMI webGUI has definitely soured my experience with Asrockrack. When I first got it it was completely unusable until it was updated!

All my Supermicro boards have been great though. Even the Java based ones!

**Comment by JayBrown** [comment]
Maybe a stupid question. One of the boards on my favorites list is the ASRock Rack W680D4U. However, I would only use it for a home server, so I probably won't need IPMI. So my question is: can you completely disable IPMI and the BMC on this board, e.g. in the BIOS? I wouldn't want the BMC & the IPMI to draw power, if I didn't use it.

**Reply by BadCo-NZ** [reply]
Not a stupid question at all. On my Supermicro boards there is a jumper for disabling the IPMI. Check your motherboard manual if Asrockrack do the same.

IPMI is super usefull though, I like checking the power consumption of all my servers from the IPMI app.

**Reply by JayBrown** [reply]
Yes, I know about the Supermicro jumper solution. Very handy. As for ASRock, the only thing I could find is a jumper to connect the PECI (Platform Environment Control Interface) to either the BMC (default setting) or the PCH. But I can't say if connecting it to the PCH would actually *disable* the BMC/IMPI sysem. (?)

![snap](https://i.imgur.com/TnKEFEp.jpg)

**Comment by BadCo-NZ** [comment]
@ironicbadger are you able to confirm if this command removes the MAC addresss for the second IPMI (eth1) port?

I noticed in the BIOS that the MAC address is missing under `BMC Network Configuration`

**Comment by smithj33** [comment]
All you have to do is uncheck the Enable LAN on the interface you don't want the BMC on. So if you want eth1 strictly for LAN/WAN functionality, uncheck Enable LAN on eth1. This doesn't physically disable eth1, it just doesn't allow the BMC to use it.
I believe what Asrock is doing, and it's weird, is linking the IPMI port to eth0 and eth1, so the IPMI is a sort of fake IPMI port. It requires one of the eth ports. Very different than SuperMicro who has an actual dedicated IPMI port that can operate without the eth ports or share eth1 to reduce cables. You always need 2 cables with Asrock.
